src/System.Net.Http/tests/FunctionalTests/HttpClientTestBase.cs

```diff
@@ -28,16 +28,17 @@ public abstract class HttpClientTestBase : RemoteExecutorTestBase
 
         protected static HttpClientHandler CreateHttpClientHandler(bool useSocketsHttpHandler)
         {
-            if (!PlatformDetection.IsNetCore) // SocketsHttpHandler only exists on .NET Core
+            if (!PlatformDetection.IsNetCore || useSocketsHttpHandler)
             {
                 return new HttpClientHandler();
             }
 
+            // Create platform specific handler.
             ConstructorInfo ctor = typeof(HttpClientHandler).GetConstructor(BindingFlags.NonPublic | BindingFlags.Instance, null, new Type[] { typeof(bool) }, null);
             Debug.Assert(ctor != null, "Couldn't find test constructor on HttpClientHandler");
 
-            HttpClientHandler handler = (HttpClientHandler)ctor.Invoke(new object[] { useSocketsHttpHandler });
-            Debug.Assert(useSocketsHttpHandler == IsSocketsHttpHandler(handler), "Unexpected handler.");
+            HttpClientHandler handler = (HttpClientHandler)ctor.Invoke(new object[] { false });
+            Debug.Assert(false == IsSocketsHttpHandler(handler), "Unexpected handler.");
```

Comment:
I'd suggest reverting the changes to this method.  They worked fine before these changes, right?  The nice thing about keeping it the way it was is that we then always check whether we've actually created the handler we wanted to.  By short-circuiting earlier, if we changed HttpClientHandler later and accidentally made it no longer use SocketsHttpHandler by default, we wouldn't discover that, or at least not as quickly.

Reply:
> They worked fine before these changes, right?

Yes, will revert the change.